### DISCUSSION TITLE
HTTPS endpoint as consumer

### DISCUSSION THREAD

**Discussion by SpuriousGer** [nats-io/nats.go#1701] [Ideas]
I'm currently looking for an alternative to Google Cloud PubSub and one of the features I've been relying on heavily is the subscription push to https endpoint. 

NATS looks very promising but am I correct in assuming that I would need to run a separate app to implement this functionality?

**Comment by heikkilamarko** [comment]
Hi. As far as I know, NATS doesn't have direct support for this.

You can very easily implement the HTTP push with [Redpanda Connect](https://docs.redpanda.com/redpanda-connect/) (Benthos) using its NATS inputs and HTTP output:

[nats](https://docs.redpanda.com/redpanda-connect/components/inputs/nats/)
[nats_jetstream](https://docs.redpanda.com/redpanda-connect/components/inputs/nats_jetstream/)
[http_client](https://docs.redpanda.com/redpanda-connect/components/outputs/http_client/)

I recommend watching this video from the NATS team about NATS + Benthos (Redpanda Connect):
[Stream processing: the easy way (with NATS and Benthos)](https://www.youtube.com/watch?v=EA2Pco3EvpU)

**Comment by jnmoyne** [comment]
Why not use the web socket transport option (i.e. use a connect URL that starts with `wss://`)? You can then just use directly the NATS clients' API and subscribe to subjects or stream consumers to get messages pushed to you in real time.
